src/sys/socket/sockopt.rs

```diff
@@ -152,6 +160,10 @@ macro_rules! sockopt_impl {
         sockopt_impl!(Both, $name, $level, $flag, usize, GetUsize, SetUsize);
     };
 
+    (Both, $name:ident, $level:path, $flag:path, OsString<$ty:ty>) => {
```

Comment:
The variable name `ty` is confusing here, because that name is also used by the `getsockopt_impl` macro.  Could you please rename it?

Reply:
Right. Done.